_meeting-notes/2019-10-22.md

```diff
@@ -0,0 +1,128 @@
+Test Ground Weekly
+===
+
+###### tags: `IPFS` `Testing`
+
+:::info
+- **Host:** @daviddias
+- **Location:** https://protocol.zoom.us/my/daviddias
+- **Date:** Oct 22, 2019 5:00 PM (UTC)
+- **Participants:**
+    - @daviddias
+    - @jimpick
+    - @stebalien
+    - @momack2
+- **Agenda**
+  1. Sprint Update `20min`
+  2. Identify blockers / Answer Questions `20min`
+  3. This week's planning `20min`
+- **Reference:** 
+  - [Last week meeting minute](https://github.com/ipfs/testground/pull/69)
+  - [Trello Board](https://trello.com/b/O9z3ljaH/testground-ignition-%F0%9F%9A%80)
+  - [Testing Infra Design](https://github.com/ipfs/testground/blob/master/docs/SPEC.md)
+  - [Testing Inventory](https://github.com/ipfs/testground/blob/master/docs/test-inventory.md)
+  - [Testing Improvement Tracker](https://docs.google.com/spreadsheets/d/1xyqyGUF-oe3x9ln88YonVeOMWWdknik74lVgL_3dBY8/edit#gid=0)
+
+:::
+
+## :mega: Update (Done, Shipped, Made Progress)
+
+- ==@daviddias==
+  - Completed the "go-ipfs as a Library" Tutorial, now in review phase by the docs Team https://github.com/ipfs/go-ipfs/pull/6695/
+  - Landed smlbench working with TestGround orchestrator (thank you Jim for helping with the last mile there ❤️) https://trello.com/c/1LTXALLW/62-get-smlbench-plan-to-work-with-testground-orchestrator
+  - Continue iterating on the Test Plan spec  https://trello.com/c/0erPZId8/43-specify-10-test-plans
+    - now with differentiation for Plan & Test Parameters https://github.com/ipfs/testground/pull/76
+    - now each Test Plan lives with the plan folder of the Plan itself https://github.com/ipfs/testground/pull/77
+  - Revisited the OKRs as requested https://trello.com/c/o09gs18G/63-review-and-iterate-on-q4-okrs
+  - Created a draft Roadmap for review https://github.com/ipfs/testground/pulls
+  - Got sick, then got better :) ETOOMUCHLONGFLIGHTS
+  - Broke a tooth 🦷 (tip: watch out for those Swiss Ricolas!). **FYI** will be traveling this week to Lisbon for a emergency dentist appt (and a reschedule of others I was going to reschedule to post hack week, but this way I merge it all together rather than going to Lisbon 2x). This means that I'm almost at 2 days of lost work given flying + transportation + being at the dentist.
+- ==@jimpick==
+  - smlbench (adds a randomly generated file on one node, gets it from a 2nd node)
+    - initial work, one docker container, IPTB with 2 IPFS nodes https://github.com/ipfs/testground/pull/74
+    - follow-on work, two dockers containers, IPTB with 1 IPFS node per container, using Sync SDK (redis) to co-ordinate https://github.com/ipfs/testground/pull/74
+  - Lots of little puzzles solved:
+    - IPFS into Docker using Docker.template and conditional via shell script (works, but not satisfied)
+    - IPTB SDK usage
+    - Sync SDK usage
+  - Tested on a single EC2 vm ... going to test on multiple VMs today
```

Comment:
@jimpick when you say you tested in EC2. Did you spin a EC2 VM from your local machine or did you install TestGround on EC2 Vm and ran it there?

Reply:
I just manually created an EC2 VM in the AWS web console and then I ssh into it, checkout out testground, and run the test.

I used a custom AMI image generated using Packer with this template: https://github.com/jimpick/testground-packer-templates

That pre-installs Docker, redis, Filebeat etc.

I've also got a few throwaway shell scripts here that might be useful...

https://github.com/jimpick/testground-misc-scripts

(shortcuts for clearing redis, cleaning up docker containers, images and networks, running various stages, no docs)

Reply:
Got it! Makes sense. How far (or are we even on track) to get the ability to spin the instances on AWS from our local machines through TestGround?

Reply:
I think that's totally possible with a small amount of extra work.

Right now, the existing local_docker runner assumes that it is "local".

I created a Trello card describing a possible ssh-based runner that captures the experimentation I've done so far - I was assuming that would run inside the first EC2 machine in the cluster, but there's no reason it couldn't run on a laptop, etc. When we implement an automated orchestrator that does CI, that will necessarily have to run in the cloud.

Currently, it's pretty easy to go into AWS directly, create an EC2 instance (ideally inside a scaling group) right from the AWS console, then install testground, configure Filebeat, and use it. I'll get some better tutorial documentation written very soon.